ibm/service/kubernetes/data_source_ibm_container_ingress_instance.go

```diff
@@ -0,0 +1,114 @@
+// Copyright IBM Corp. 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package kubernetes
+
+import (
+	"fmt"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceIBMContainerIngressInstance() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMContainerIngressInstanceRead,
+		Schema: map[string]*schema.Schema{
+			"instance_crn": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Instance CRN id",
+			},
+			"cluster": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Cluster ID",
+				ValidateFunc: validate.InvokeDataSourceValidator(
+					"ibm_container_ingress_instance",
+					"cluster"),
+			},
+			"instance_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Instance registration name",
+			},
+			"secret_group_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: "Secret group for the instance registration",
+			},
+			"is_default": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Designates if the instance is the default for the cluster",
+			},
```

Comment:
these two fields look to be computed ones. I think these should have Computed:true, instead Optional and Default.

Reply:
Do you mind going into more detail about what "computed" is? Or what the difference between optional and compute is? Reading the description, it seems like computed is if that's something the backend can extrapolate. For both of these fields, they are optional but the user can pass in an option. Secret Group ID is optional, and if not passed in will default to "". If is default is not passed in, it will default to false.

Reply:
`is_default` and `secret_group_id` are not computed fields, they are optionally specified by the user

Reply:
I talked to @z0za about it offline

Reply:
is_default and secret_group_id are computed right as part of datasource we are not making any call like d.Get() on those fields. They acts as computed for datasources

Reply:
yes you are right, for gets they are computed. I updated them to be marked as such